clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -363,6 +363,19 @@ public void testConfigProvidersPropsAsParam() {
         assertEquals(config.originals().get("sasl.kerberos.password"), "randomPassword");
     }
 
+    @Test
+    public void testImmutableOriginalsWithConfigProvidersProps() {
```

Comment:
Does this test fail without the change?

Reply:
@tadsul, what is the answer to this question? (Unresolving this conversation.)

Reply:
yes the test fails without the comment

Reply:
Hmm, I am not sure it would. You should run the test without the changes to `AbstractConfig.java` to verify. The test invokes `new TestIndirectConfigResolution(props, immutableMap)` where neither `props` nor `immutableMap` is actually immutable. In order fot the test to fail `props` needs to be unmodifiable since that is the map used as `originals`. And `immutableMap` needs to be renamed since it is confusing  as it is neither immutable nor the map being tested for immutability.

Reply:
oh yes i had to modify the test to make it work .. my yes meant that i could reproduce the issue.. sorry about that i will post the modified test